### ISSUE TITLE
Dynamic core crash on reset

### ISSUE THREAD

**Issue by rderooy** [joncampbell123/dosbox-x#2215]
@koolkdev I was testing with the dynamic core, and when Win98SE asked to restart the system due to some changes, it crashed DOSBox-X. This is with the latest nightly code. But looking at the log message this seems like a known issue.

These where the last messages in the log:
....
LOG: APM BIOS: OS disconnected
LOG: Plug & Play OS reports itself inactive
LOG: Restart by keyboard controller requested
LOG: Warning: C++ exception method is not compatible with dynamic core when emulating reset
terminate called after throwing an instance of 'int'
Aborted (core dumped)
....

**Comment by Wengier** [comment]
@koolkdev probably wants to take a look at this. Thanks!

**Comment by koolkdev** [comment]
It isn't a new thing, but now that I have added support for catching exceptions in dynamic core, I can add support for catching int exceptions. I think that adding that support for IO_WriteB would be enough since it seems to be the only place in the code that int exception is thrown from the core itself. (hence that warning)

**Comment by joncampbell123** [comment]
@koolkdev You are correct about that. As far as I know all methods to reset the CPU are I/O based.

I highly doubt there's anything out there that allows resetting the CPU through memory mapped I/O.

**Comment by rderooy** [comment]
I tried an install of Win98SE in DOSBox-X nightly, and this issue happens during installation when WIndows reboots.

I had changed the Windows installation instructions on the draft wiki, from core=normal to core=dynamic_x86 as default (with a note to go back to core=normal if you run into problems), but now I'm thinking this was too early and that I need to revert it to core=normal, at least for the installation process.

**Comment by rderooy** [comment]
For now I have changed the default win9x config files back to ``core=normal``, with a note that users can switch to ``core=dynamic_x86`` after installation is finished. At least until this issue is resolved.

**Comment by rderooy** [comment]
Just a small comment that simply starting WinME with ``core=dynamix_x86`` (after installing it with ``core=normal``), causes dosbox-x to core dump on bootup.

**Comment by ajalberd** [comment]
@rderooy Can Windows 98 be installed + run on CPU higher than pentium_mmx type? Such as pentium_ii?

**Comment by WinCollecter** [comment]
This is good, thanks for helping me understand this issue @rderooy!
